### ISSUE TITLE
Transformscale puts polygon outside of bounds from original one

### ISSUE THREAD

**Issue by Philip2909** [Turfjs/turf#2107]
Hello! First of all I wanna thank everyone who has worked on this project and to all the people who decided to make it public for anyone to use, as it is a lifesaver and a really good tool, but onto my problem. I have buildings, houses I am drawing in 3d but it does not really matter as another thing handles the hight problem, so basicly its 2d. I have a geojson (Polygon) of a house/building, with its corners and to this building I want to add walls that have a width and the way I did that was with turfs transformscale and then mask, to get an outer wall. This works great on almost all buildings and houses but it has a bit of a problem with houses with wierd shapes. It puts the scaled down version of the polygon outside of the bounds of the house, if that makes any sense, and that makes it just fill all the space that should be "open". I am gonna add some pictures that shows my problem.

![image](https://user-images.githubusercontent.com/51290462/122574403-e94c0800-d04f-11eb-9433-c87c545ca1bf.png)
Now here i put the transformscale to 0.5 to show it more dramaticly (usually I use 0.95), but this was just to show a point.

![image](https://user-images.githubusercontent.com/51290462/122574650-29ab8600-d050-11eb-884a-b0418b76d2b7.png)
Weird rendering glitch I was talking about, red one is the mask (scaled down one) and the white is the result of the mask one. (turf.mask).

Is my request even possible to do? If so what can I do, or is there something on your end you guys could do? 

TL;DR
Transformscale does not put the scaled down polygon within bounds of the polygon you gave it, is it possible to fix somehow?

Thanks for all help in advance!

- Version [6.3.0]

**Comment by matkoniecz** [comment]
looking at http://turfjs.org/docs/#transformScale it should be solvable by manually specifying origin if centroid is unsuitable.

(Or maybe there is a bug in centroid calculation? But more likely is that centroid is not suitable for such scaling)

**Comment by Philip2909** [comment]
Hello! I also tested it out before by changing the origin but now I went and tried out all of them but non of them did what I wanted it to do, I also see that 'center' and 'centroid' has no differance, should that be the case or is it bugged? Do you have any idea of how to make a scaling system that keeps the scaled down version "inside" of the original bbox?

**Comment by rowanwins** [comment]
Hi @Philip2909 , 
I think unfortunately it's because of the shape of your object that you are having trouble. In theory you can pass in a custom coordinate to the origin parameter, so if you only have to do it as a one off that might be achievable to work it out manually. Or perhaps you could try using [centerOfMass](http://turfjs.org/docs/#centerOfMass) and see if that generates a different point from `centroid` or `center`?
For further help you probably need to share your geojson feature.

Cheers
Rowan

**Comment by Philip2909** [comment]
Hi Rowan!
I don't know if I understood it wrong but how would it help to do centerOfMass?  You mean I pass the origon as the centerofmass? I am sorry but I don't understand what you mean. I am gonna link my polygon I am having problems with, remember I only get the corners of a building, and from that have to make "walls" if you can call it that. 

https://gist.github.com/Philip2909/ad513c25a67708faa51a698afc86b8be

Best Regards
Philip

**Comment by Philip2909** [comment]
I figured out what you meant with center of mass, sadly that did not work as I wanted, I manually placed a point and then checked if it was good as the origin, and I found this to work really good: 
https://gist.github.com/Philip2909/cb212476019e40182a2d59dfa845e670

Is it possible to get a point like this somehow?

**Comment by stebogit** [comment]
@Philip2909 I don't think there is a definition for the point you suggested.
Now, this is just a guess, I wonder if you would get good results shrinking your polygons in multiple steps using the center of mass, so that the origin "moves" with each transformation. In other words, if you want to shrink your polygon to say 80% of it's original size, then you transform it like 10 times by 2%.
My assumption, which might be just wrong, is that the location of the center of mass of each smaller polygon does not coincide, in a concave polygon, with the center of mass of its "parent". 🤔

**Comment by Philip2909** [comment]
Hi, I tried your idea out, I guess it would work on some buildings with weird shapes that the original one did not work on but sadly it did not cut it on this one. I think it does not move the center of mass too far for it to be effective. I tried all kinds of combinations, a lot of times with very little scaling or a fewer amount of times and more scaling.

**Comment by rowanwins** [comment]
Hi @Philip2909 

I think you're probably out of luck with what turf can do out of the box, there is no one approach to the centoid-ish issue that suits all irregular polygons. I'm going to close this issue but if you ever find a neat solution feel free to post back in the future.

Thanks,
Rowan
